access/grpc/grpc.go

```diff
@@ -1188,14 +1188,10 @@ func (c *BaseClient) SubscribeBlocksFromStartBlockID(
 		return nil, nil, newRPCError(err)
 	}
 
-	blocksChan := make(chan flow.Block)
-	errChan := make(chan error)
-
-	go func() {
-		defer close(blocksChan)
-		defer close(errChan)
-		receiveBlocksFromClient(ctx, subscribeClient, blocksChan, errChan)
-	}()
+	convertBlockResponse := func(response *access.SubscribeBlocksResponse) (flow.Block, error) {
+		return convert.MessageToBlock(response.GetBlock())
+	}
+	blocksChan, errChan := subscribe(ctx, subscribeClient.Recv, convertBlockResponse, "block")
```

Comment:
or even something like this if you switch the error to use type
```suggestion
	return subscribe(ctx, subscribeClient.Recv, func(response *access.SubscribeBlocksResponse) (flow.Block, error) {
		return convert.MessageToBlock(response.GetBlock())
	})
```

Reply:
I actually like giving a name to a convert function as it looks more readable rather than an unnamed callback

Reply:
I mean change `subscribe` to return `(<-chan Response, <-chan error, error)`, where the error is always nil